wasmedge/executor.go

```diff
@@ -0,0 +1,136 @@
+package wasmedge
+
+// #include <wasmedge/wasmedge.h>
+import "C"
+import (
+	"runtime"
+	"unsafe"
+)
+
+type Executor struct {
+	_inner *C.WasmEdge_ExecutorContext
+	_own   bool
+}
+
+func NewExecutor() *Executor {
+	executor := C.WasmEdge_ExecutorCreate(nil, nil)
+	if executor == nil {
+		return nil
+	}
+	res := &Executor{_inner: executor, _own: true}
+	runtime.SetFinalizer(res, (*Executor).Release)
+	return res
+}
+
+func NewExecutorWithConfig(conf *Configure) *Executor {
+	executor := C.WasmEdge_ExecutorCreate(conf._inner, nil)
+	if executor == nil {
+		return nil
+	}
+	res := &Executor{_inner: executor, _own: true}
+	runtime.SetFinalizer(res, (*Executor).Release)
+	return res
+}
+
+func NewExecutorWithStatistics(stat *Statistics) *Executor {
+	executor := C.WasmEdge_ExecutorCreate(nil, stat._inner)
+	if executor == nil {
+		return nil
+	}
+	res := &Executor{_inner: executor, _own: true}
+	runtime.SetFinalizer(res, (*Executor).Release)
+	return res
+}
+
+func NewExecutorWithConfigAndStatistics(conf *Configure, stat *Statistics) *Executor {
+	executor := C.WasmEdge_ExecutorCreate(conf._inner, stat._inner)
+	if executor == nil {
+		return nil
+	}
+	res := &Executor{_inner: executor, _own: true}
+	runtime.SetFinalizer(res, (*Executor).Release)
+	return res
+}
+
+func (self *Executor) Instantiate(store *Store, ast *AST) error {
+	res := C.WasmEdge_ExecutorInstantiate(self._inner, store._inner, ast._inner)
+	if !C.WasmEdge_ResultOK(res) {
+		return newError(res)
+	}
+	return nil
+}
+
+func (self *Executor) RegisterImport(store *Store, imp *ImportObject) error {
+	res := C.WasmEdge_ExecutorRegisterImport(self._inner, store._inner, imp._inner)
+	if !C.WasmEdge_ResultOK(res) {
+		return newError(res)
+	}
+	return nil
+}
+
+func (self *Executor) RegisterModule(store *Store, ast *AST, modname string) error {
+	modstr := toWasmEdgeStringWrap(modname)
+	res := C.WasmEdge_ExecutorRegisterModule(self._inner, store._inner, ast._inner, modstr)
+	if !C.WasmEdge_ResultOK(res) {
+		return newError(res)
+	}
+	return nil
+}
+
+func (self *Executor) Invoke(store *Store, funcname string, params ...interface{}) ([]interface{}, error) {
+	funcstr := toWasmEdgeStringWrap(funcname)
+	funccxt := store.FindFunction(funcname)
+	ftype := funccxt.GetFunctionType()
+	cparams := toWasmEdgeValueSlide(params...)
```

Comment:
Does `cparams` need to be deleted at the end of this function?

Reply:
`WasmEdge_Value` not needs to be deleted.